### PR TITLE
Switch to go modules and update dependencies

### PR DESCRIPTION
**Pull request by saschagrunert**
I also updated the CI scripts to check vendoring.

### CODE REVIEW

**Comment by saschagrunert** [comment]
I saw that we have to drop support for go < 1.12, but this should not harm.

**Comment by vrothberg** [comment]
> I saw that we have to drop support for go < 1.12, but this should not harm.

Do you know why? Go 1.12 might not be available for all users (especially in the enterprise world).

**Comment by saschagrunert** [comment]
> > I saw that we have to drop support for go < 1.12, but this should not harm.
> 
> Do you know why? Go 1.12 might not be available for all users (especially in the enterprise world).

Because we restrict it with the go.mod file. I would assume that CRI-O and podman are (hopefully) the only consumers of this project, right? Sooo it should not harm but feel free to correct me.

**Comment by vrothberg** [comment]
> Because we restrict it with the go.mod file. I would assume that CRI-O and podman are (hopefully) the only consumers of this project, right? Sooo it should not harm but feel free to correct me.

`go build` will ignore the module when `GO111MODULE=off` or when using  `-mod=vendor`. The `go 1.12` line in `go.mod` doesn't seem to impose a requirement either, as it can be used with v1.11. In other words, I don't see why this PR drops support go < 1.12. If it would, I would object strongly as 1.11 is still in use, even on development machines.

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,7 @@ language: go
 dist: trusty
 
 go:
-  - 1.9.x
```
Comment (vrothberg) [inline .travis.yml]: Could we add `1.11.x` as well? Just to be sure that it's working.

Comment (saschagrunert) [reply .travis.yml]: I readded all previous versions and added 1.12.

Comment (saschagrunert) [reply .travis.yml]: And now 1.11, too

**Pull request merged**